src/Symfony/Component/DependencyInjection/MemoizeProxy/BaseKeyGenerator.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection\MemoizeProxy;
+
+class BaseKeyGenerator implements KeyGeneratorInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function __invoke(string $className, string $method, array $arguments): string
+    {
+        return hash('sha256', $className.$method.serialize($arguments));
```

Comment:
Just remember. You don't know what is inside in $arguments variable. May be anything, for example a closure instance that's impossible to serialize here.

Reply:
I totally agree. The `KeyGenerator` is configurable to allow user to handle this kind of specific cases.